pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -46,8 +46,8 @@ def __init__(
     ) -> None:
         self._model: Optional[Module] = None
         checkpoint_io = checkpoint_io if checkpoint_io is not None else TorchCheckpointIO()
-        self._checkpoint_io = checkpoint_io
-        self._precision_plugin = precision_plugin if precision_plugin is not None else PrecisionPlugin()
+        self.checkpoint_io = checkpoint_io
+        self.precision_plugin = precision_plugin if precision_plugin is not None else PrecisionPlugin()
```

Comment:
We need this to be self._precision_plugin instead of direct set to self.precision_plugin
[In 3/n move accelerator into strategy](https://github.com/PyTorchLightning/pytorch-lightning/pull/11022) The accelerator_connector depends on distinguishing is user set up a non_None precision plugin or we set default to decide whether call _select_precision_plugin(), so self._precision_plugin is what user passed in, and self.precision_plugin is default value

Reply:
```suggestion
        self._checkpoint_io = checkpoint_io
        self._precision_plugin = precision_plugin
```
```suggestion
        self.checkpoint_io = checkpoint_io
        self.precision_plugin = precision_plugin
```

Reply:
Add a issue to discuss future 
https://github.com/PyTorchLightning/pytorch-lightning/issues/11097

Reply:
Do the current changes fix #11097?